### PR TITLE
Add tasks filters for v0.30

### PR DESCRIPTION
**Pull request by alallema**
Add new filters on the `get /tasks` route.

Present in this [spec](https://github.com/meilisearch/specifications/pull/195)

V0.30.0
in TasksQuery:
  - [x] Add `uid`
  - [x] Add `CanceledBy`
  - [x] Add `BeforeEnqueuedAt`
  - [x] Add `AfterEnqueuedAt`
  - [x] Add `BeforeStartedAt`
  - [x] Add `AfterStatedAt`
  - [x] Add `BeforeFinishedAt`
  - [x] Add `AfterFinishedAt`

Tests:
  - [x] `uid`
  - [x] `canceledBy`
  - [x] `beforeEnqueuedAt`

### CODE REVIEW

File: client_test.go [hunk 1]
```diff
@@ -747,9 +747,51 @@ func TestClient_GetTasks(t *testing.T) {
...
+			name: "TestGetTasksWithUidFilter",
+			args: args{
+				UID:    "indexUID",
+				client: defaultClient,
+				document: []docTest{
+					{ID: "123", Name: "Pride and Prejudice"},
+				},
+				query: &TasksQuery{
+					Limit: 1,
+					UIDs:  []int64{1},
```
Comment (brunoocasali) [inline client_test.go:766]: following the same idea of the `IndexUIDs` the `UIDs` could be just `uids`.

File: client.go [hunk 2]
```diff
@@ -390,9 +377,7 @@ func convertKeyToParsedKey(key Key) (resp KeyParsed) {
 	// Convert time.Time to *string to feat the exact ISO-8601
 	// format of Meilisearch
 	if !key.ExpiresAt.IsZero() {
-		const Format = "2006-01-02T15:04:05"
```
[the remark this answers is not in the document]
Comment (alallema) [reply client.go:393]: I don't know any other way except [Format method](https://pkg.go.dev/time#Time.Format)

[the remark this answers is not in the document]
Comment (brunoocasali) [reply client.go:393]: I mean, in other languages is very common to have `YYYY-MM-DD` or something like this, but in this case you're using a real date value to do the format...

[the remark this answers is not in the document]
Comment (alallema) [reply client.go:393]: Yes I know ...

**Pull request merged**